### PR TITLE
Rewrite

### LINKED ISSUE #4: Undefined variable: filetype
Getting this Notice when accessing the Gitium/commits page:

PHP Notice:  Undefined variable: filetype in /www/wp-content/plugins/autover/autover.php on line 197 when calling autover_version_filter(), request: GET /wp-admin/admin.php?page=gitium/gitium-commits.php, request_id: b886268f9382083fbf7e23e2db1bef44

### PR DESCRIPTION
**Pull request by krodyrobi**
Fixes #4

### CODE REVIEW

File: tests/test-autover.php [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		];
+	}
+
+	/**
+	 * @dataProvider data_url_to_path
+	 */
+	public function test_version_acquisition_from_proper_file( $url, $expected_path ) {
+		$filemtime_called = false;
+		redefine( 'filemtime', function ( $file_path ) use ( $expected_path, &$filemtime_called ) {
+			$this->assertSame( $expected_path, $file_path );
```
Comment (t-ionut) [inline tests/test-autover.php]: Same opinion as for the last time you put an assertion in your test env setup, drop it

File: tests/test-autover.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+			$filemtime_called = true;
+
+			return 9999;
+		} );
+
+		$is_file_called = false;
+		redefine( 'is_file', function ( $file_path ) use ( $expected_path, &$is_file_called ) {
+			$this->assertSame( $expected_path, $file_path );
```
Comment (t-ionut) [inline tests/test-autover.php]: Same suggestions as a few lines above

File: src/readme.txt [hunk 3]
```diff
@@ -54,25 +53,30 @@ The 'admin_*' hooks are designed for wp-admin area or so called Dashboard area,
 
 == Changelog ==
 
+= 1.5 =
+* WP-CLI skeleton
```
Comment (uibar) [inline src/readme.txt]: WP-CLI?? dose does not matter if it was generated using wp-cli. maybe it's better to just say something like "upgrade repo structure to match recommended structure."

File: src/readme.txt [hunk 4]
```diff
@@ -54,25 +53,30 @@ The 'admin_*' hooks are designed for wp-admin area or so called Dashboard area,
...
+* Tests added
+* cleanup of code
```
Comment (uibar) [inline src/readme.txt]: Use capital letters at every point like bellow.

File: tests/test-autover.php (hunk 1 above)
Comment (t-ionut) [inline tests/test-autover.php]: If you need to assert a variable that doesn't get back to you, then just make another flag like
```php
redefine( 'filemtime', function ( $file_path ) use ( $expected_path, &$received_path, &$filemtime_called ) {
    $received_path = $file_path;
    $filtemtime_called = true;

    return 9999;
} );

// blah blah

$this->assertSame( $expected_path, $received_path );
```

File: tests/test-autover.php [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+use function Patchwork\restoreAll;
+
+
+class AutoverTest extends WP_UnitTestCase {
+	public function tearDown() {
+		restoreAll();
+		parent::tearDown();
+	}
+
+	public function test_sample() {
```
Comment (uibar) [inline tests/test-autover.php]: test_sample? really? what's the value of this?

File: src/autover.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Plugin Name: Autover
+ * Plugin URI: http://wordpress.org/extend/plugins/autover/
+ * Description: Automatically version your CSS and JS files enqueued through the Wordpress API
+ * Author: Presslabs
+ * Version: 1.5
+ * Author URI: http://www.presslabs.com/
```
Comment (krodyrobi) [inline src/autover.php]: Got it the first time :D.

Comment (t-ionut) [inline src/autover.php]: :imp:

**Comment by t-ionut** [comment]
LGTM, just squash those commits into one or two meaningful ones (e.g. rewrite & tests)

**Pull request merged**